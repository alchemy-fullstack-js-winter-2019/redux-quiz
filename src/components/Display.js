import React from 'react';
import PropTypes from 'prop-types';
import styles from './app/App.css';

function Display({
  title,
  subtitle
}) {
  return (
    <div className={styles.display}>
      <h1>{title}</h1>
      <h2>{subtitle}</h2>
    </div>
  );
}

Display.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
};

export default Display;
