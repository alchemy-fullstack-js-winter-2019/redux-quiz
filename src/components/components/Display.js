import React from 'react';
import PropTypes from 'prop-types';

function Display({ title, subtitle }) {
  return (
    <>
      <h1> {title} </h1>
      <h3> {subtitle} </h3>
    </>
  );
}

Display.propTypes =  {
  title: PropTypes.string,
  subtitle: PropTypes.string
};

export default Display;
