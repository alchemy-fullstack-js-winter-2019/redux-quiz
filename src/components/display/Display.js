import React from 'react';
import PropTypes from 'prop-types';

export default function Display({ title, subtitle }) {
    return (
        <>
            <h1>{title}</h1>
            <h2>{subtitle}</h2>
        </>
    );
}

Display.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
};
