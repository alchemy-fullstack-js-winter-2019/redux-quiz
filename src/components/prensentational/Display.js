import React from 'react';
import PropTypes from 'prop-types';

function Display({ title, subTitle }) {
  return (
    <>
      <h1> {title} </h1>
      <h2> {subTitle} </h2>
    </>
  );
}

Display.propTypes =  {
  title: PropTypes.string,
  subTitle: PropTypes.string
};

export default Display;
