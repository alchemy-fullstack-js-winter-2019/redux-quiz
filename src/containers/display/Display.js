import { connect } from 'react-redux';
import { getTitle, getSubtitle } from '../../selectors/Controls';
import Display from '../../components/display/Display';

const mapStateToProps = state => ({
    title: getTitle(state),
    subtitle: getSubtitle(state)
});


export default connect(mapStateToProps)(Display);
