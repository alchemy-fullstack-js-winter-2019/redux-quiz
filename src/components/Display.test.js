import React from 'react';
import renderer from 'react-test-renderer';
import Display from './Display';

describe('display snapshot', () => {
  it('can match a snapshot', () => {
    const tree = renderer.create(
      <Display
        title='hello'
        subtitle='goodbye'
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
