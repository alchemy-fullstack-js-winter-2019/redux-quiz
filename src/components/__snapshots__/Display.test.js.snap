// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display snapshot can match a snapshot 1`] = `
<div>
  <h1>
    hello
  </h1>
  <h2>
    goodbye
  </h2>
</div>
`;
